src/Value/Locale.php

```diff
@@ -52,7 +52,8 @@ public static function fromString($string): self
 
     public function equals(self $locale): bool
     {
-        return 0 === \strcmp((string) $this, (string) $locale);
+        return $this->regionCode->equals($locale->regionCode) &&
```

Comment:
Did you also attempt to just do a `return (string) $this === (string) $locale` instead? I would expect that to result in an even higher performance boost.

Reply:
I actually tried to avoid all the `__toString` (and underlying `sprintf`) calls. So no, did not try that, but could give it a go next month 🙂